common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java

```diff
@@ -41,6 +42,32 @@ public static Double value(GridCoverage2D rasterGeom, Geometry geometry, int ban
         return values(rasterGeom, Collections.singletonList(geometry), band).get(0);
     }
 
+    public static Double value(GridCoverage2D rasterGeom, int colX, int rowY, int band) throws TransformException {
+        int numBands = rasterGeom.getNumSampleDimensions();
+        if (band < 1 || band > numBands) {
+            // Invalid band index. Return nulls.
+            return null;
+        }
```

Comment:
Please use the RasterUtil.ensureBand method for this

Reply:
Using RasterUtil.ensureBand will throw an IllegalArgumentException. For RS_Value we expect to return null for invalid bands.

Reply:
Is there a reason for it?